src/PageContent/RequestPickup/RecurringPickupRequest.js

```diff
@@ -101,12 +101,12 @@ class RecurringPickupRequest extends Component {
             errors['startDate'] = 'Start date cannot be empty';
         }
 
-        if (!fields['durationType']) {
+        if (!fields['endCriteria']) {
             // the they need to have one or the other error msg
-            errors['durationType'] = 'Must select radio button';
+            errors['endCriteria'] = 'Must select radio button';
             formIsValid = false;
         } else {
-            if (fields['durationType'] === RequestDurationType.DATE) {
+            if (fields['endCriteria'] === RequestEndCriteriaType.DATE) {
```

Comment:
Leon's actually right here. Because inside that last `else` is another `if`, so it should really be 
```
if (!fields['endCriteria']) {
  ...
} else if (fields['endCriteria'] === RequestEndCriteriaType.DATE) {
  ...
} else if(fields['occurTimes'] === '' || !fields['occurTimes'] || fields['occurTimes'] < 2) {
  ...
}
```

fwiw I was just string replacing here, so I take no responsibility for this conditional madness hahaha  but @leontaolong very very nice catch, makes the code much cleaner to unnest things.

Reply:
ah i see, i didn't see the other `if`. Good call!